lib/cacheman.ex

```diff
@@ -190,7 +190,7 @@ defmodule Cacheman do
   def fetch(name, key, put_opts, fallback) do
     case get(name, key) do
       {:ok, nil} ->
-        case fallback.() do
+        case fallback.(key) do
```

Comment:
🤔 what do you think about adding a guard for the fallback?
something like this:
```elixir
# for anonymous fallbacks
def fetch(name, key, put_opts, fallback) when is_function(fallback, 0)
# for single argument fallbacks
def fetch(name, key, put_opts, fallback) when is_function(fallback, 1) 
```

This way we won't break existing API, that relies on zero-argument fallbacks

Reply:
You can even guard in the case statement: 

```elixir
{:ok, nil} when is_function(fallback, 0) ->
        case fallback.()
        
{:ok, nil} when is_function(fallback, 1) ->
        case fallback.(key)
```

Reply:
Great idea, didn't know that was a possibility. I added `is_function` check within the `case`